service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java

```diff
@@ -222,18 +224,19 @@ public static HttpResponse executeGetRequest(@Nonnull String url) throws IOExcep
     List<BasicNameValuePair> getNameValuePairsWithExpandedLists(Map<String, Object> parameters) {
         return parameters.entrySet().stream()
                 .flatMap(entry -> {
+                    String key = KEY_MAPPING.getOrDefault(entry.getKey(), entry.getKey());
```

Comment:
I don't understand why you add this KEY_MAPPING to this method? What is the expected result you want to see returned?

Reply:
His mapping should map the customfield name "paymentData" to the name of the payone request parameter "add_paydata[action}". I cannot use "add_paydata[action]" as customfieldname, because "][" are not allowed in the customfield names,